### PR TITLE
Edge test app in kotlin

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

- Adds a new app-kotlin target for testing Edge SDKs in Kotlin. The app provides similar functionality with the java test app.
- Moves Commerce XDM formatters to a util library app-util-xdm to be shared across test apps.

## Screenshots (if appropriate):

Screen Shot 2023-01-29 at 7 47 36 PM
Screen Shot 2023-01-29 at 7 47 45 PM
Screen Shot 2023-01-29 at 7 48 09 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: code/app-util-xdm/build.gradle [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    compileOptions {
+        sourceCompatibility rootProject.ext.sourceCompatibility
+        targetCompatibility rootProject.ext.targetCompatibility
+    }
+}
+
+dependencies {
+    implementation project(':edge')
```
Comment (addb) [inline code/app-util-xdm/build.gradle:33]: do we need edge as a dependency here?

Comment (addb) [inline code/app-util-xdm/build.gradle:33]: If we move this xdm https://github.com/adobe/aepsdk-edge-android/blob/dev-v2.0.0/code/edge/src/main/java/com/adobe/marketing/mobile/xdm/Property.java
 to the tester package we can remove edge project dependency right?

File: code/app-kotlin/src/main/java/com/adobe/marketing/tester/app/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+package com.adobe.marketing.tester.app
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.*
+import androidx.appcompat.app.AppCompatActivity
+import com.adobe.marketing.mobile.*
```
Comment (addb) [inline code/app-kotlin/src/main/java/com/adobe/marketing/tester/app/MainActivity.kt:21]: Will this also import all the edge classes?

File: code/app-kotlin/src/main/java/com/adobe/marketing/tester/app/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        findViewById<Button>(R.id.btnResetIdentities).setOnClickListener {
+            MobileCore.resetIdentities()
+        }
+
+        findViewById<Button>(R.id.btnSubmitSingleEvent).setOnClickListener {
+
+            val valuesList: MutableList<String> = ArrayList()
+            valuesList.add("val1")
+            valuesList.add("val2")
+            val eventData: MutableMap<String, Any> = java.util.HashMap()
```
Comment (addb) [inline code/app-kotlin/src/main/java/com/adobe/marketing/tester/app/MainActivity.kt:193]: let's use kotlin `mapOf<String , String>()`?

**Review by addb — APPROVED** [review]
Left few nit comments. Overall it  looks good!

File: code/app-util-xdm/build.gradle (hunk 1 above)
Comment (emdobrin) [reply code/app-util-xdm/build.gradle:33]: yes, for Schema and Property references

Comment (emdobrin) [reply code/app-util-xdm/build.gradle:33]: we cannot move it, is part of the Edge SDK

Comment (addb) [reply code/app-util-xdm/build.gradle:33]: Move as in create a copy of it in the tester package.

Comment (emdobrin) [reply code/app-util-xdm/build.gradle:33]: Does not make much sense to copy them. The Edge extension provides them for Schema implementors, hence the dependency on Edge SDK for this util lib.

**Review by addb — APPROVED** [review]

**Pull request merged**